### PR TITLE
feat: Initial project structure

### PR DESCRIPTION
**Pull request by Woynert**
Includes:

- Single .JAR file building with `shadowJar` plugin.
- Testing with Junit Jupiter (Junit5).
- Coverage reports with Jacoco.
- Simple SOAP service exposed at `http://localhost:8080/sampleservice?wsdl`.
- Dockerfile & docker-compose.
- Code formatting with `clang-format` see `./format.sh`.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.jar
Comment (shoriwe) [inline gradle/wrapper/gradle-wrapper.jar:1]: Does this binary file is really necessary?

File: docker-compose.yml
Comment (shoriwe) [inline docker-compose.yml:1]: The `docker-compose.yaml` in this repo will setup the other services. So `gateway` developers can start their environment easily.

**Review by shoriwe — CHANGES_REQUESTED** [review]

File: gradle/wrapper/gradle-wrapper.jar
Comment (Woynert) [reply gradle/wrapper/gradle-wrapper.jar:1]: According to the Gradle documentation

> To make the Wrapper files available to other developers and execution environments you’ll need to check them into version control. All Wrapper files including the JAR file are very small in size. Adding the JAR file to version control is expected. Some organizations do not allow projects to submit binary files to version control. At the moment there are no alternative options to the approach.

https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:adding_wrapper

File: docker-compose.yml
Comment (Woynert) [reply docker-compose.yml:1]: Removed

**Pull request merged**